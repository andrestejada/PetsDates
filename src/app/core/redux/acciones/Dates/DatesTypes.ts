import { Citas} from '../../../../feature/Citas/interfaces/index';

export const ADD_DATE = 'ADD_DATE';
export const GET_ALL_DATES = 'GET_ALL_DATES';
export const DELETE_DATE = 'DELETE_DATE';

interface AccionAddDate {
  type: typeof ADD_DATE;
  payload: Citas;  
}
interface AccionGetDates {
  type: typeof GET_ALL_DATES;
  payload: Citas[];  
}

interface AcctionDeleteDate{
  type: typeof DELETE_DATE,
  payload:number
}
interface AccionDefaultState {  
  type: ''
}


export type DatesDispatchTypes  =
  | AcctionDeleteDate
  | AccionAddDate
  | AccionGetDates
  | AccionDefaultState;
 