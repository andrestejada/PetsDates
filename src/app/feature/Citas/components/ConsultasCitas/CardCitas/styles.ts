import styled from 'styled-components';

export const TextInformation = styled.span`
    color: #000;
    font-weight: bold;
    text-transform: capitalize;
`;
export const ButtonContainer = styled.div`
    display: flex;
    gap: 12px;
    padding: 0.5em 0;
`;
