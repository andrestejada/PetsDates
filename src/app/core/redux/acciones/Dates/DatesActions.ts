import { ADD_DATE,DELETE_DATE, DatesDispatchTypes, GET_ALL_DATES,  } from './DatesTypes';
import { Cita,Citas} from 'app/feature/Citas/interfaces';
import { Dispatch } from 'redux';
import { axiosIntance } from '../../../config/AxiosConfig';

export const SaveDate=(date:Cita)=>{
    return async (dispatch:Dispatch<DatesDispatchTypes>)=>{
        try {
            const {data} = await  axiosIntance.post('/dates',date);
            dispatch( addNewDate(data) );
        } catch (error) {
            console.log(error);
        }
    };
};


export const addNewDate=(data:Citas):DatesDispatchTypes=>({
    type:ADD_DATE,
    payload:data
});


export const GetDates=()=>{
    return async (dispatch:Dispatch<DatesDispatchTypes>)=>{
        try {
            const {data} = await  axiosIntance.get('/dates');
            dispatch( getAllDates(data) );
        } catch (error) {
            console.log(error);
        }
    };
};

export const getAllDates=(data:Citas[]):DatesDispatchTypes=>({
    type:GET_ALL_DATES,
    payload:data
});


export const deleteDateByID=(id:number)=>{
    return async(dispatch:Dispatch<DatesDispatchTypes>)=>{
        try {
            await axiosIntance.delete(`/dates/${id}`);
            dispatch(deleteDate(id));
        } catch (error) {
            console.log(error);
        }
    };
};


export const deleteDate=(id:number):DatesDispatchTypes=>({
    type:DELETE_DATE,
    payload:id
});