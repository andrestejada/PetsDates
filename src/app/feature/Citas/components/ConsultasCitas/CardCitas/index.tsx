import { Button, Card, CardContent, Typography } from '@material-ui/core';
import { ButtonContainer, TextInformation } from './styles';
import { Citas } from '../../../interfaces/index';
import DeleteIcon from '@material-ui/icons/Delete';
import EditIcon from '@material-ui/icons/Edit';
import React from 'react';



const CardCitas = ({
  id,
  nombreMascota,
  nombrePropietario,
  tarifa,
  tipoServicio,
  fechaHora,
  observaciones
}: Citas) => {
  return (
    <>
      <Card>
        <CardContent>
          <Typography>
            Nombre de la Mascota:
            <TextInformation>{nombreMascota}</TextInformation>
          </Typography>
          <Typography>
            Nombre del Propietario:
            <TextInformation>{nombrePropietario}</TextInformation>
          </Typography>
          <Typography>
            Tipo de servicio
            <TextInformation> {tipoServicio} {`($${tarifa})`}</TextInformation>
          </Typography>
          <Typography>
            Fecha y Hora:
            <TextInformation>{fechaHora}</TextInformation>
          </Typography>
          <Typography>
            Observaciones:
            <TextInformation>{observaciones}</TextInformation>
          </Typography>
          <ButtonContainer>
            <Button
              variant="contained"
              color="secondary"
              startIcon={<DeleteIcon />}
            >
              Borrar
            </Button>
            <Button
              variant="contained"
              color="primary"
              startIcon={<EditIcon />}
            >
              Editar
            </Button>
          </ButtonContainer>
        </CardContent>
      </Card>
    </>
  );
};

export default CardCitas;
