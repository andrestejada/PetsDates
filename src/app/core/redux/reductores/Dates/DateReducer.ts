import { ADD_DATE, DELETE_DATE, DatesDispatchTypes, GET_ALL_DATES } from '../../acciones/Dates/DatesTypes';
import { Citas } from '../../../../feature/Citas/interfaces/index';

export interface Dates{
    allDates:Citas[]
}
const initialState:Dates={
    allDates:[]
};


export const dateReducer=(state:Dates=initialState,action:DatesDispatchTypes):Dates=>{
    switch (action.type) {
        case ADD_DATE:            
            return{
                ...state,
                allDates:[...state.allDates,action.payload]
            };    
        case GET_ALL_DATES:
            return{
                ...state,
                allDates:action.payload
            };
        case DELETE_DATE:
            return{
                ...state,
                allDates: state.allDates.filter( (cita)=> cita.id !== action.payload)
            };
        default:
            return state;
    }
};