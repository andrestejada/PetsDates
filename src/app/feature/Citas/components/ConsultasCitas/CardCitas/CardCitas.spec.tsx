import CardCitas from './index';
import { Provider } from 'react-redux';
import React from 'react';
import createMockStore from 'redux-mock-store';
import { deleteDateByID } from '../../../../../core/redux/acciones/Dates/DatesActions';
import { mount } from 'enzyme';
import thunk from 'redux-thunk';
const middlewares = [thunk]; // add your middlewares like `redux-thunk`
const mockStore = createMockStore(middlewares);

const initialState = {};
const store = mockStore(initialState);
store.dispatch = jest.fn();

jest.mock('../../../../../core/redux/acciones/Dates/DatesActions', () => ({
  deleteDateByID: jest.fn(),
}));

describe('testing <CardCitas/> component', () => {
  let compponentWrappe = mount(
    <Provider store={store}>
      <CardCitas
        id={12345}
        nombrePropietario={'Andres'}
        nombreMascota={'Sol'}
        tipoServicio={'basico'}
        tarifa={10000}
        fechaHora={'2021-12-01T16:00'}
        observaciones={'pelo corto'}
      />
    </Provider>
  );

  beforeEach(() => {
    compponentWrappe = mount(
      <Provider store={store}>
        <CardCitas
          id={12345}
          nombrePropietario={'Andres'}
          nombreMascota={'Sol'}
          tipoServicio={'basico'}
          tarifa={10000}
          fechaHora={'2021-12-01T16:00'}
          observaciones={'pelo corto'}
        />
      </Provider>
    );
  });

  it('should be call the deleteDateByID', () => {
    compponentWrappe.find('button').simulate('click');
    expect(deleteDateByID).toHaveBeenCalled();
  });
  it('should be call the deleteDateByID with the arguments', () => {
    compponentWrappe.find('button').simulate('click');
    expect(deleteDateByID).toHaveBeenCalledWith(12345);
  });
});
