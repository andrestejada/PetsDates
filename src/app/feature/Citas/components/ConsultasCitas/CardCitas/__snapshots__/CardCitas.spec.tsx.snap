// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <CardCitas/> component shoul be render correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      Nombre de la Mascota:
      <span
        className="sc-AxjAm ccbxoJ"
      >
        Sol
      </span>
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      Nombre del Propietario:
      <span
        className="sc-AxjAm ccbxoJ"
      >
        Andres
      </span>
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      Tipo de servicio
      <span
        className="sc-AxjAm ccbxoJ"
      >
         
        basico
         
        ($10000)
      </span>
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      Fecha y Hora:
      <span
        className="sc-AxjAm ccbxoJ"
      >
        2021-12-01T16:00
      </span>
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      Observaciones:
      <span
        className="sc-AxjAm ccbxoJ"
      >
        pelo corto
      </span>
    </p>
    <div
      className="sc-AxirZ jdhWsB"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          <span
            className="MuiButton-startIcon MuiButton-iconSizeMedium"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
            </svg>
          </span>
          Borrar
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
